CMakeLists.txt

```diff
@@ -87,209 +87,106 @@ endfunction()
-  GIT_TAG        "${tag}"
-  PREFIX         "${CMAKE_BINARY_DIR}/nemsio"
-  LIST_SEPARATOR |
-  CMAKE_ARGS     "-DCMAKE_PREFIX_PATH=${CMAKE_PREFIX_PATH_ALT_SEP}"
-                 "-DCMAKE_INSTALL_PREFIX=${CMAKE_INSTALL_PREFIX}/nemsio/nemsio-${ver}"
-                 "-DENABLE_TESTS=${ENABLE_TESTS}"
-  DEPENDS        bacio w3nco
-  )
-
-getTagVer("${LIBCOMPS_FILE}" "nemsiogfs" tag ver)
-ExternalProject_Add(nemsiogfs
-  GIT_REPOSITORY "https://github.com/noaa-emc/nceplibs-nemsiogfs"
-  GIT_TAG        "${tag}"
-  PREFIX         "${CMAKE_BINARY_DIR}/nemsiogfs"
-  LIST_SEPARATOR |
-  CMAKE_ARGS     "-DCMAKE_PREFIX_PATH=${CMAKE_PREFIX_PATH_ALT_SEP}"
-                 "-DCMAKE_INSTALL_PREFIX=${CMAKE_INSTALL_PREFIX}/nemsiogfs/nemsiogfs-${ver}"
-                 "-DENABLE_TESTS=${ENABLE_TESTS}"
-                 "-DOPENMP=${OPENMP}"
-  DEPENDS        nemsio
-  )
-
-getTagVer("${LIBCOMPS_FILE}" "landsfcutil" tag ver)
-ExternalProject_Add(landsfcutil
-  GIT_REPOSITORY "https://github.com/noaa-emc/nceplibs-landsfcutil"
-  GIT_TAG        "${tag}"
-  PREFIX         "${CMAKE_BINARY_DIR}/landsfcutil"
-  LIST_SEPARATOR |
-  CMAKE_ARGS     "-DCMAKE_PREFIX_PATH=${CMAKE_PREFIX_PATH_ALT_SEP}"
-                 "-DCMAKE_INSTALL_PREFIX=${CMAKE_INSTALL_PREFIX}/landsfcutil/landsfcutil-${ver}"
-                 "-DENABLE_TESTS=${ENABLE_TESTS}"
-  DEPENDS        nemsio
-  )
-
-getTagVer("${LIBCOMPS_FILE}" "g2" tag ver)
-ExternalProject_Add(g2
-  GIT_REPOSITORY "https://github.com/noaa-emc/nceplibs-g2"
-  GIT_TAG        "${tag}"
-  PREFIX         "${CMAKE_BINARY_DIR}/g2"
-  LIST_SEPARATOR |
-  CMAKE_ARGS     "-DCMAKE_PREFIX_PATH=${CMAKE_PREFIX_PATH_ALT_SEP}"
-                 "-DCMAKE_INSTALL_PREFIX=${CMAKE_INSTALL_PREFIX}/g2/g2-${ver}"
-                 "-DENABLE_TESTS=${ENABLE_TESTS}"
-  )
-
-getTagVer("${LIBCOMPS_FILE}" "g2tmpl" tag ver)
-ExternalProject_Add(g2tmpl
-  GIT_REPOSITORY "https://github.com/noaa-emc/nceplibs-g2tmpl"
-  GIT_TAG        "${tag}"
-  PREFIX         "${CMAKE_BINARY_DIR}/g2tmpl"
-  LIST_SEPARATOR |
-  CMAKE_ARGS     "-DCMAKE_PREFIX_PATH=${CMAKE_PREFIX_PATH_ALT_SEP}"
-                 "-DCMAKE_INSTALL_PREFIX=${CMAKE_INSTALL_PREFIX}/g2tmpl/g2tmpl-${ver}"
-                 "-DENABLE_TESTS=${ENABLE_TESTS}"
-  )
+set(nemsio_depends bacio sigio)
+set(nemsiogfs_depends nemsio)
+set(landsfcutil_depends nemsio)
+set(w3emc_depends nemsio sigio)
+
+foreach(lib ${nceplibs})
+  getTagVer("${LIBCOMPS_FILE}" "${lib}" tag ver)
+  if(DEPLOY)
+    set(lib_install ${lib}/${lib}-${ver})
+  else()
+    set(lib_install ${lib}-${ver})
+  endif()
+  
+  ExternalProject_Add(${lib}
+    GIT_REPOSITORY "https://github.com/noaa-emc/nceplibs-${lib}"
+    GIT_TAG        "${tag}"
+    PREFIX         "${CMAKE_BINARY_DIR}/${lib}"
+    LIST_SEPARATOR |
+    CMAKE_ARGS     "-DCMAKE_PREFIX_PATH=${CMAKE_PREFIX_PATH_ALT_SEP}"
+    "-DCMAKE_INSTALL_PREFIX=${CMAKE_INSTALL_PREFIX}/${lib_install}"
+    "-DENABLE_TESTS=${ENABLE_TESTS}"
```

Comment:
Fix indentation

Reply:
Is that indentation not good? That's the way emacs `cmake-mode` auto formats it.

Reply:
I guess that is OK for cmake, just not visually appealing.

Reply:
I indented it